libc/cmake/modules/LLVMLibCCompileOptionRules.cmake

```
@@ -262,13 +263,13 @@ function(_get_common_test_compile_options output_var c_test flags)
       list(APPEND compile_options "-Wno-gnu-imaginary-constant")
     endif()
     list(APPEND compile_options "-Wno-pedantic")
-    # if(NOT CMAKE_COMPILER_IS_GNUCXX)
-    #   list(APPEND compile_options "-Wnewline-eof")
-    #   list(APPEND compile_options "-Wnonportable-system-include-path")
-    #   list(APPEND compile_options "-Wstrict-prototypes")
-    #   list(APPEND compile_options "-Wthread-safety")
-    #   list(APPEND compile_options "-Wglobal-constructors")
-    # endif()
+    if(NOT CMAKE_COMPILER_IS_GNUCXX)
+      list(APPEND compile_options "-Wnewline-eof")
+      list(APPEND compile_options "-Wnonportable-system-include-path")
+      list(APPEND compile_options "-Wstrict-prototypes")
+      list(APPEND compile_options "-Wthread-safety")
+      # list(APPEND compile_options "-Wglobal-constructors") # triggered in TEST_F implementation
```

Right, so I would adjust this commit description so that it doesn't close #119281 as resolved. Instead, I would link to the issue from a comment like so:

```suggestion
      # TODO(https://github.com/llvm/llvm-project/issues/119281) triggered in TEST_F implementation
      # list(APPEND compile_options "-Wglobal-constructors")
```

---

Fixed